compiler/rustc_infer/src/traits/mod.rs

```diff
@@ -129,6 +129,9 @@ pub enum FulfillmentErrorCode<'tcx> {
     CodeSubtypeError(ExpectedFound<Ty<'tcx>>, TypeError<'tcx>), // always comes from a SubtypePredicate
     CodeConstEquateError(ExpectedFound<Const<'tcx>>, TypeError<'tcx>),
     CodeAmbiguity,
+    /// Overflow reported from the new solver `-Ztrait-solver=next`, which will
+    /// be reported as an regular error as opposed to a fatal error.
+    CodeOverflow,
```

Comment:
can you merge `CodeOverflow` with `CodeAmbiguity`?

This should make it clearer that overflow should be treated as ambiguous and not as an actual hard error.

So something like `CodeAmbuguity { overflow: bool }`

r=me after that